sphinx/templates/latex/longtable.tex_t

```diff
@@ -10,25 +10,51 @@
 <%- if table.caption -%>
 \sphinxthelongtablecaptionisattop
 \caption{<%= ''.join(table.caption) %>\strut}<%= labels %>\\*[\sphinxlongtablecapskipadjust]
+<%- if table.booktabs -%>
```

Comment:
I'm thinking about new templates set is needed for booktabs because these if-block makes templates complex. I worry that it will be hard to support yet another table style.

Reply:
A possibility would be to use in the template `\sphinxtoprule`, `\sphinxmidrule`, `\sphinxbottomrule` with `\def\sphinxtoprule{\hline}` etc... and other definitions with booktabs triggering option.